### PR TITLE
HDDS-8881. Add more BlockDeletingServiceMetrics to Monitor the Datanode

### PR DESCRIPTION
**Pull request by xichen01**
## What changes were proposed in this pull request?

Add more BlockDeletingServiceMetrics to Monitor the Datanode

## What is the link to the Apache JIRA

```bash
block_deleting_service_metrics_failure_count{context="dfs",hostname="VM-8-3-centos"} 0
block_deleting_service_metrics_marked_block_count{context="dfs",hostname="VM-8-3-centos"} 5210
block_deleting_service_metrics_out_of_order_delete_block_transaction_count{context="dfs",hostname="VM-8-3-centos"} 1
block_deleting_service_metrics_received_block_count{context="dfs",hostname="VM-8-3-centos"} 5210
block_deleting_service_metrics_received_container_count{context="dfs",hostname="VM-8-3-centos"} 20
block_deleting_service_metrics_received_retry_transaction_count{context="dfs",hostname="VM-8-3-centos"} 0
block_deleting_service_metrics_received_transaction_count{context="dfs",hostname="VM-8-3-centos"} 28
block_deleting_service_metrics_success_bytes{context="dfs",hostname="VM-8-3-centos"} 16498688
block_deleting_service_metrics_success_count{context="dfs",hostname="VM-8-3-centos"} 4028
block_deleting_service_metrics_total_block_chosen_count{context="dfs",hostname="VM-8-3-centos"} 4028
block_deleting_service_metrics_total_container_chosen_count{context="dfs",hostname="VM-8-3-centos"} 16
block_deleting_service_metrics_total_pending_block_count{context="dfs",hostname="VM-8-3-centos"} 0
```
new metrics

```bash
block_deleting_service_metrics_marked_block_count{context="dfs",hostname="VM-8-3-centos"} 5210
block_deleting_service_metrics_received_block_count{context="dfs",hostname="VM-8-3-centos"} 5210
block_deleting_service_metrics_received_container_count{context="dfs",hostname="VM-8-3-centos"} 20
block_deleting_service_metrics_received_retry_transaction_count{context="dfs",hostname="VM-8-3-centos"} 0
block_deleting_service_metrics_received_transaction_count{context="dfs",hostname="VM-8-3-centos"} 28
block_deleting_service_metrics_total_block_chosen_count{context="dfs",hostname="VM-8-3-centos"} 4028
block_deleting_service_metrics_total_container_chosen_count{context="dfs",hostname="VM-8-3-centos"} 16
```

## How was this patch tested?

manual test and some metrics have integration test

### CODE REVIEW

File: hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/common/helpers/BlockDeletingServiceMetrics.java [hunk 1]
```diff
@@ -86,6 +108,34 @@ public void incrFailureCount() {
     this.failureCount.incr();
   }
 
+  public void incrReceiveTransactionCount(long count) {
+    receivedTransactionCount.incr(count);
+  }
+
+  public void incrReceiveRetryTransactionCount(long count) {
```
Comment (szetszwo) [inline hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/common/helpers/BlockDeletingServiceMetrics.java:115]: Typo: it should be "incr`Received`RetryTransactionCount".

**Review by szetszwo — COMMENTED** [review]
@xichen01 , thanks for working on this!  The change looks good.  Just a few comment inlined.

File: hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/common/helpers/BlockDeletingServiceMetrics.java (hunk 1 above)
Comment (xichen01) [reply hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/common/helpers/BlockDeletingServiceMetrics.java]: done.

File: hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/common/helpers/BlockDeletingServiceMetrics.java [hunk 2]
```diff
@@ -86,6 +108,34 @@ public void incrFailureCount() {
     this.failureCount.incr();
   }
 
+  public void incrReceiveTransactionCount(long count) {
```
[the remark this answers is not in the document]
Comment (xichen01) [reply hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/common/helpers/BlockDeletingServiceMetrics.java]: done.

File: hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/common/helpers/DeletedContainerBlocksSummary.java [hunk 3]
```diff
@@ -34,6 +34,8 @@ public final class DeletedContainerBlocksSummary {
   // key : txID
   // value : times of this tx has been processed
   private final Map<Long, Integer> txSummary;
+  private final Map<Long, Integer> retryTxSummary;
```
[the remark this answers is not in the document]
Comment (xichen01) [reply hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/common/helpers/DeletedContainerBlocksSummary.java]: done.

**Review by szetszwo — APPROVED** [review]
+1 the change looks good.

**Comment by adoroszlai** [comment]
Thanks @xichen01 for the patch, @szetszwo for the review.

**Pull request merged**